panther_analysis_tool/enriched_event_generator.py

```diff
@@ -0,0 +1,180 @@
+GNU Affero General Public License for more details.
+
+You should have received a copy of the GNU Affero General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+"""
+
+import logging
+from tkinter import E
+
+from ruamel.yaml import CommentedMap as YAMLCommentedMap
+
+from panther_analysis_tool.analysis_utils import LoadAnalysisSpecsResult
+from panther_analysis_tool.backend.client import Client as BackendClient
+from panther_analysis_tool.backend.client import GenerateEnrichedEventParams
+from panther_analysis_tool.constants import AnalysisTypes
+
+TEST_CASE_FIELD_KEY_LOG = "Log"
+TEST_CASE_FIELD_KEY_RESOURCE = "Resource"
+
+
+class EnrichedEventGenerator:
+    """Enriches test data for analysis items."""
+
+    def __init__(self, backend: BackendClient):
+        """Initializes the TestDataEnricher.
+
+        Args:
+            backend: Backend API client.
+        """
+        self.backend = backend
+
+    @staticmethod
+    def _filter_analysis_items(
+        analysis_items: list[LoadAnalysisSpecsResult],
+    ) -> list[LoadAnalysisSpecsResult]:
+        """Filters analysis items to only those that need test data enrichment.
+
+        Args:
+            analysis_items: A list of analysis items to filter.
+
+        Returns:
+            A list of analysis items that have the Tests property and an AnalysisType of RULE, POLICY, or SCHEDULED_RULE.
+        """
+        return [
+            item
+            for item in analysis_items
+            if item.analysis_spec.get("Tests")
+            and item.analysis_spec["AnalysisType"]
+            in [AnalysisTypes.RULE, AnalysisTypes.POLICY, AnalysisTypes.SCHEDULED_RULE]
+        ]
+
+    @staticmethod
+    def _convert_inline_json_dict_to_python_dict(data: YAMLCommentedMap) -> dict:
+        """Converts YAML-loaded inline JSON to Python dictionaries. This allows them
+        to be re-serialized into YAML instead of maintaining JSON formatting.
+        """
+        if isinstance(data, dict):
+            new_dict = {}
+            for key, value in data.items():
+                if isinstance(value, (dict, list)):
+                    new_dict[key] = EnrichedEventGenerator._convert_inline_json_dict_to_python_dict(
+                        value
+                    )
+                else:
+                    new_dict[key] = value
+            return new_dict
+        elif isinstance(data, list):
+            new_list = []
+            for item in data:
+                if isinstance(item, (dict, list)):
+                    new_list.append(
+                        EnrichedEventGenerator._convert_inline_json_dict_to_python_dict(item)
+                    )
+                else:
+                    new_list.append(item)
+            return new_list
+        return data
+
+    def _handle_analysis_item(self, analysis_id: str, test: dict, test_case_field_key: str) -> dict:
+        if test_case_field_key not in test:
```

Comment:
we should also check to see if the event CAN be enriched.  IIRC a p_log_type field is required on the event for it to be enriched. We can probably pre-populate that if we aren't already doing that (just gets a bit tricky with a rule that has multiple log types)

Reply:
yeah we may need that then. The way you currently have the logic in place a test might get modified even if it wasn't enriched. Are you doing any special checks for that scenario?